### DISCUSSION TITLE
Not being able to restore hub to lego firmware

### DISCUSSION THREAD

**Discussion by sunnysideup121** [pybricks/support#1987] [Q&A]
I tried restoring the Lego firmware on the spike prime hubs which has the pybrick firmware installed. The Bluetooth button blinks all the colors - pink, green, blue, off but when I clicked the restore button it my hub doesn't show up in the window to restore. Can someone help? I followed the instructions on the site.

**Comment by dlech** [comment]
Usually, if the hub doesn't show up, it is due to a bad USB cable. Some USB cables are for charging only and can't be used for sending data.

Does the hub show up in the Windows Device manager?

**Reply by sunnysideup121** [reply]
I'm using the cable that was provide with Spike Prime. I also don't see the hub on Devise Manager. What kind of cable do I need to make this work?

**Reply by dlech** [reply]
The one that comes in the LEGO set should is fine, assuming it hasn't been damaged. But if nothing is showing up at all in the Windows Device Manager when you unplug the hub and plug it back in, then something strange is going on at a hardware level.

If the driver isn't installed, you should see this:
![image](https://github.com/user-attachments/assets/b0bc4dd7-2a3a-462c-a7fa-0dd627fff7ed)

But if you managed to install Pybricks firmware using the same computer, the driver should still be configured to be using the WinUSB driver and look like this:
![image](https://github.com/user-attachments/assets/956b9eed-dc44-426f-971d-e94ba325623f)

**Comment by laurensvalk** [comment]
Can you try it on the same computer you've used to install Pybricks originally?

It should work on any computer, but certainly on the one you used previously, because the installation and recovery procedures work exactly the same.

**Reply by ziyidu** [reply]
Hello, excuse me, my computer should have Pybricks firmware installed, because my driver is configured to use the WinUSB driver, the one in your picture, there is no warning sign, but when I download the firmware or restore, it says "No compatible device found"
